### PR TITLE
bumper, Change bumper fail policy

### PR DESCRIPTION
**Pull request by RamLavi**
**What this PR does / why we need it**:
This PR is opened in relation to a issue #693 opened due to knmstate bump fail that caused the bumper to stop bumping.

- **bumper, Change bumper fail policy**
Currently, the bumper script will fail upon the first bump failure.
We want to be able to run the bumper on all components even if one
component bump is broken.
The new behavior is that all found bumps will occur, and in the end if at least one
component failed to bump - the job will exit with error.

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by RamLavi** [comment]
/override pull-e2e-cluster-network-addons-operator-nmstate-functests
irrelevant to PR

File: tools/bumper/bumper.go [hunk 1]
```diff
@@ -82,12 +84,17 @@ func main() {
...
+				logger.Printf("Bump %s component was initiated but did not succeed. err = %s", componentName, err)
+				failedComponents = append(failedComponents, componentName)
 			}
 		} else {
 			logger.Printf("Bump not needed in component %s", componentName)
 		}
 	}
+
+	if len(failedComponents) != 0 {
+		exitWithError(fmt.Errorf("Not all component bumps were successful: %s", strings.Join(failedComponents, ",")))
```
Comment (qinqon) [inline tools/bumper/bumper.go:96]:
```suggestion
		exitWithError(fmt.Errorf("Not all component bumps were successful, failed components: %s", strings.Join(failedComponents, ",")))
```

**Review by qinqon — CHANGES_REQUESTED** [review]
nit

File: tools/bumper/bumper.go (hunk 1 above)
Comment (RamLavi) [reply tools/bumper/bumper.go]: Done

**Pull request merged**